client/state/sites/plans/selectors.js

```diff
@@ -46,10 +47,44 @@ export const getCurrentPlan = ( state, siteId ) => {
 		debug( 'current plan: %o', plan );
 		return plan;
 	}
-
 	return null;
 };
 
+/**
+ * Returns a site specific plan
+ * @param  {Object} state        global state
+ * @param  {Number} siteId       the site id
+ * @param  {String} productSlug  the plan product slug
+ * @return {Object} the matching plan
+ */
+export const getSitePlan = createSelector(
+	( state, siteId, productSlug ) => {
+		const plansBySiteId = getPlansBySiteId( state, siteId );
+		if ( ! plansBySiteId || plansBySiteId.isRequesting || ! plansBySiteId.data ) {
+			return null;
+		}
+		return plansBySiteId.data.filter( plan => plan.productSlug === productSlug ).shift();
+	},
+	( state, siteId ) => getPlansBySiteId( state, siteId )
+);
+
+/**
+ * Returns a plan discount price
+ * @param  {Object}  state         global state
+ * @param  {Number}  siteId       the site id
+ * @param  {String}  productSlug   the plan product slug
+ * @param  {Boolean} isMonthly     if true, returns monthly price
+ * @return {Number}  plan discount price
+ */
+export function getPlanDiscountPrice( state, siteId, productSlug, isMonthly = false ) {
+	const plan = getSitePlan( state, siteId, productSlug );
+	if ( get( plan, 'rawPrice', -1 ) < 0 || get( plan, 'rawDiscount', -1 ) <= 0 ) {
+		return null;
+	}
+	const discountPrice = plan.rawPrice;
+	return isMonthly ? parseFloat( ( discountPrice / 12 ).toFixed( 2 ) ) : discountPrice;
```

Comment:
Hmm, okay. I won't lobby for this at the moment as everybody has their own preferences and it would be pretty complicated to set some hard rules about vertical spacing I think.

Feel free to leave it as it is. It was just a suggestion. :)